idaes/models_extra/gas_solid_contactors/unit_models/tests/test_mb_performance.py

```diff
@@ -0,0 +1,242 @@
+from pyomo.dae.flatten import flatten_dae_components
+from pyomo.environ import (
+    ConcreteModel,
+    check_optimal_termination,
+    SolverStatus,
+    value,
+    Var,
+    Constraint,
+)
+from pyomo.util.check_units import assert_units_consistent
+from pyomo.common.config import ConfigBlock
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+from idaes.core import (
+    FlowsheetBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    MomentumBalanceType,
+)
+from idaes.core.util.model_statistics import (
+    degrees_of_freedom,
+    number_variables,
+    number_total_constraints,
+    number_unused_variables,
+    unused_variables_set,
+    large_residuals_set,
+)
+from idaes.core.util.testing import initialization_tester
+from idaes.core.util import scaling as iscale
+from idaes.core.solvers import get_solver
+from idaes.core.util.exceptions import InitializationError
+
+# Import MBR unit model
+from idaes.models_extra.gas_solid_contactors.unit_models.moving_bed import MBR
+
+# Import property packages
+from idaes.models_extra.gas_solid_contactors.properties.methane_iron_OC_reduction.gas_phase_thermo import (
+    GasPhaseParameterBlock,
+)
+from idaes.models_extra.gas_solid_contactors.properties.methane_iron_OC_reduction.solid_phase_thermo import (
+    SolidPhaseParameterBlock,
+)
+from idaes.models_extra.gas_solid_contactors.properties.methane_iron_OC_reduction.hetero_reactions import (
+    HeteroReactionParameterBlock,
+)
+
+
+def build_model(
+    dynamic=False,
+    #ntfe=10,
+    #ntcp=3,
+):
+    m = ConcreteModel()
+    if dynamic:
+        m.fs = FlowsheetBlock(
+            default={
+                "dynamic": dynamic,
+                "time_set": [0, 600],
+                "time_units": pyo.units.s,
+            },
+        )
+    else:
+        m.fs = FlowsheetBlock(default={"dynamic": dynamic})
+
+    # Set up thermo props and reaction props
+    m.fs.gas_properties = GasPhaseParameterBlock()
+    m.fs.solid_properties = SolidPhaseParameterBlock()
+
+    m.fs.hetero_reactions = HeteroReactionParameterBlock(
+        default={
+            "solid_property_package": m.fs.solid_properties,
+            "gas_property_package": m.fs.gas_properties,
+        }
+    )
+
+    m.fs.MB = MBR(
+        default={
+            "has_holdup": True,
+            "transformation_method": "dae.collocation",
+            "gas_phase_config": {"property_package": m.fs.gas_properties},
+            "solid_phase_config": {
+                "property_package": m.fs.solid_properties,
+                "reaction_package": m.fs.hetero_reactions,
+            },
+        }
+    )
+    return m
+
+
+def initialize_model(m, dynamic=False, ntfe=10, ntcp=3):
+    if dynamic:
+        disc = pyo.TransformationFactory("dae.collocation")
+        disc.apply_to(
+            m, wrt=m.fs.time, nfe=ntfe, ncp=ntcp, scheme="LAGRANGE-RADAU"
+        )
+
+    # Fix bed geometry variables
+    m.fs.MB.bed_diameter.fix(6.5)  # m
```

Comment:
In line with the comment above, I generally include fixing variables as part of model construction rather than initialization (including a test for degrees of freedom). Would there be an issue with moving this code there?

Reply:
The geometry variables can be fixed in `build_model` without issue, but the dynamic inputs must be fixed after time discretization. The only reason time discretization happens in `initialize_model` is to avoid unit inconsistency when using the `TestModel._run_test` method.

Reply:
So, two things here:

1. The time discretization should occur in the `build_model` step, as it is part of model construction (and thus we want o include the time there).
2. Regarding the unit consistency issue, for now we should be skipping this step for dynamic models due to the known Pyomo issue (or maybe doing a `check_unit_consistency` instead and asserting that it is False`).